### PR TITLE
Add feature flags to OData $count endpoints

### PR DESCRIPTION
**Pull request by joelverhagen**
Related to https://github.com/NuGet/Engineering/issues/2902
Address https://github.com/NuGet/Engineering/issues/3442

### CODE REVIEW

File: src/NuGetGallery/Controllers/ODataV1FeedController.cs [hunk 1]
```diff
@@ -253,6 +272,38 @@ public IHttpActionResult GetPropertyFromPackages(string propertyName, string id,
...
+        {
+            return (await Search(
+                options,
+                searchTerm,
+                targetFramework,
+                _featureFlagService.IsODataV1SearchCountNonHijackedEnabled()))
+                .FormattedAsCountResult<V1FeedPackage>();
+        }
+
+        private async Task<IHttpActionResult> Search(
```
Comment (loic-sharma) [inline src/NuGetGallery/Controllers/ODataV1FeedController.cs:302]: Nit: can this be renamed to `SearchAsync`?

Comment (joelverhagen) [reply src/NuGetGallery/Controllers/ODataV1FeedController.cs:302]: Yeah good call

File: src/NuGetGallery/Controllers/ODataV2FeedController.cs [hunk 2]
```diff
@@ -60,6 +60,31 @@ public class ODataV2FeedController
...
+            [FromUri] string semVerLevel = null)
+        {
+            return (await Get(
+                options,
+                semVerLevel,
+                _featureFlagService.IsODataV2GetAllCountNonHijackedEnabled()))
+                .FormattedAsCountResult<V2FeedPackage>();
+        }
+
+        private async Task<IHttpActionResult> Get(
```
Comment (loic-sharma) [inline src/NuGetGallery/Controllers/ODataV2FeedController.cs:84]: Nit: can we also add the `Async` suffix here?

File: src/NuGetGallery/Controllers/ODataV1FeedController.cs (hunk 1 above)
Comment (loic-sharma) [reply src/NuGetGallery/Controllers/ODataV1FeedController.cs:302]: Same comment for other async private methods you added

Comment (loic-sharma) [reply src/NuGetGallery/Controllers/ODataV1FeedController.cs:302]: Ah actually I realize that you didn't moved these methods, not added them. Never mind

**Review by loic-sharma — APPROVED** [review]

File: src/NuGetGallery/Controllers/ODataV1FeedController.cs (hunk 1 above)
Comment (joelverhagen) [reply src/NuGetGallery/Controllers/ODataV1FeedController.cs:302]: The private methods are new and can be easily renamed.

**Pull request merged**